src/inputs/EditableLabel/EditableLabel.js

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { TextField, makeStyles, Typography, CircularProgress } from '@material-ui/core';
+import EditIcon from '@material-ui/icons/Edit';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    height: '100%',
+  },
+  labelDisplayContainer: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'flex-start',
+    alignItems: 'stretch',
+  },
+  label: {
+    maxWidth: '500px',
+    color: theme.palette.primary.main,
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+  },
+  textField: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'flex-start',
+    alignItems: 'stretch',
+    '& .MuiOutlinedInput-input': {
+      borderWidth: '2px',
+      outline: 'none',
+      padding: '8px',
+    },
+    '& .MuiOutlinedInput-root': {
+      minWidth: '300px',
+    },
+  },
+  editIcon: {
+    marginRight: '10px',
+    fontWeight: 'bold',
+    height: '20px',
+    color: theme.palette.primary.main,
+  },
+  circularLoading: {
+    marginLeft: '15px',
+    color: theme.palette.primary.contrastText,
+    backgroundColor: theme.palette.background.card,
+  },
+}));
+
+export const EditableLabel = (props) => {
+  const { value, onProcess, checkValue, labels } = props;
+
+  const classes = useStyles();
+
+  const [editMode, setEditMode] = useState(false);
+  const [error, setError] = useState(null);
+  const [isMouseHover, setMouseHover] = useState(false);
+  const [textFieldValue, setTextFiedlValue] = useState(value);
+
+  const startEdition = (event) => {
+    event.stopPropagation();
+    event.preventDefault();
+    setEditMode(true);
+  };
+
+  const modifyTextField = (event) => {
+    const newValue = event.target.value;
+    if (newValue !== value) {
+      setError(checkValue(newValue));
+    }
+  };
+
+  const processNewValue = (event) => {
```

Comment:
It looks like this function is only called when the users exits the text field edition, and processes the current value that is not necessarily a "new" value.
Could we rename this function to `stopEdition` ? This would mirror the `startEdition` function and it would be more intuitive when we call it in the `onBlur` and `onKeyPress` props of the TextField